big-int/src/operators/multiply.cpp

```diff
@@ -35,17 +35,38 @@ namespace libbig
 {
     largeInt largeInt::operator*(const largeInt &next_number) {
 
-        auto append_zeroes = [](const largeInt &x, const int factor) {
-            int k;
+        using Iterator = std::string::reverse_iterator; 
+
+        auto remove_zeroes = [] (const largeInt &x) {
             largeInt Answer = x;
-            for(k = 0; k < factor; k++) {
+            std::reverse(Answer.number.begin(), Answer.number.end());
+            for(Iterator j=Answer.number.rbegin(); j != Answer.number.rend() - 1; ++j) {
+                if((*j) == '0') {
+                    Answer.number.pop_back();
+                }
+                else {
+                    break;
+                }
+            }
+            std::reverse(Answer.number.begin(), Answer.number.end());
+            return Answer;
+        };
+
+        auto append_zeroes = [] (const largeInt &x, const int factor) {
+            largeInt Answer = x;
+            for(int k = 0; k < factor; k++) {
                 Answer.number.push_back('0');
             }
 
             return Answer;
         };
 
-        auto simple_multiplication = [](const largeInt &x, const largeInt &y) {
+        auto simple_multiplication = [] (const largeInt &x, const largeInt &y) {
+
+            auto char_int_converter = [] (const char &x) {
```

Comment:
let this be in some global scope, so that this can also act as an helper in other implementations